### DISCUSSION TITLE
Help with LARS for Barlow Twins

### DISCUSSION THREAD

**Discussion by dmandair** [Lightning-AI/pytorch-lightning#11535] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi and thanks in advance for the help! This is more of a general question but I'm unsure of how to use this LARS wrapper (https://pytorch-lightning-bolts.readthedocs.io/en/0.2.1/api/pl_bolts.optimizers.lars_scheduling.html) with the configure_optimizer function in Pytorch lightening. Would I just wrap an Adams optimizer with this and that would be the sole item returned by the function? Additionally in Barlow twins, the LR had a warm-up scheduling - would this need to be implemented separately or is this taken care of in the implementation or the LARS wrapper? Thanks again for the help!

**Comment by ananyahjha93** [comment]
If you are using LARS wrapper, you need to wrap an optimizer within it. However, now on the [master branch](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/optimizers/lars.py), we removed the LARS wrapper and instead added LARS, which you can use directly.

You'll have to add the warm-up schedule separately from the optimizer.

**Reply by dmandair** [reply]
Oh awesome, do I just need to 'pip install lightning-bolts' to be able to use that? Schedueler-wise, I was planning on using one similar to what's implemented here (https://pytorch-lightning.readthedocs.io/en/latest/notebooks/lightning_examples/barlow-twins.html). That should work right?
